Feature/contacts

bortoletti-giorgia opened the pull request:
## Premise

Hi, in this feature I have added some options for the contacts.
Now you can swipe to right on a contact and modify it or swipe to left and delete it.
With deletion there is also the possibility to UNDO.
You can also do research in the contacts list for names and addresses.

After a swipe for background color and icon, I have imported class `RecyclerViewSwipeDecorator` from [this github project](https://github.com/xabaras/RecyclerViewSwipeDecorator) and added implementation on `build.gradle` .

## Changes

Activities:
- To create contact
- To modificy contact
- To view contacts in a RecyclerView with searching view

Classes to manage RecyclerView:
- SMSContactSwipeCallback to manage the swipe
- SMSContactRecyclerAdapter to manage contacts list with research, deletion, creation and updating
- ContactFilter to filter contacts
- SMSContactSearchListener to invoke filter after a search request

## Side Changes

- Singleton for SMSContactManager (also done by @turcato-niccolo )
- UI with RelativeLayout

bortoletti-giorgia committed:
Updated equals method

bortoletti-giorgia committed:
Merge branch 'development' into feature/contacts

bortoletti-giorgia committed:
Added checks for validity of a new contact before insertion

bortoletti-giorgia committed:
Merge branch 'feature/design-UI' into feature/contacts

bortoletti-giorgia committed:
From LinearLayout to RelativeLayout

In the activities:
- activity_main
- activity_create_contact

bortoletti-giorgia committed:
feat: added first implementation to swipe and delete a contact

I have used a decorator for the delete swipe to see red background and the delete icon. For this I have added the implementation in the Gradle file for the object RecyclerViewSwipeDecorator from https://github.com/xabaras/RecyclerViewSwipeDecorato.

bortoletti-giorgia committed:
refactor(UI): add Toolbar in home screen

Added an AppBar that contains a Toolbar that contains a Menu, all in the home screen.

bortoletti-giorgia committed:
feat: add 'undo' for a contact just deleted

Added a Snackbar to undo the recently action of deleting a contact.

bortoletti-giorgia committed:
refactor(UI): add Toolbar in contacts list screen

Added a Toolbar with a button to search for a contact in the list below.

bortoletti-giorgia committed:
feat: implemented the search option in the contact list

User writes a text in the SearchView of the ContactListActivity. Text could be part of a name of a contact or his address. Research isn't case sensitive.
The ContactSearchListener actives onQueryTextChange method that invokes the ContactFilter.
ContactFilter takes care of filtering the data then passes them to the adapter that displays them.

bortoletti-giorgia committed:
feat: added implementation to swipe and modify a contact

Now user can swipe to left and delete contact selected, or swipe to right and modify that contact.

bortoletti-giorgia committed:
style: formatting, specs, color of the icons

bortoletti-giorgia committed:
fix: use Singleton in SMSContactManager class

bortoletti-giorgia committed:
feat: add some interfaces with GenericContact

Interfaces added:
- ContactManager to manage GenericContact
- ContactRecyclerHelper so SMSContactRecyclerAdapter has to extend Adapter and implement this interface to manage items in the view
- ContactConverterUtils to convert GenericContact and generic Peer

bortoletti-giorgia committed:
Merge branch 'development' into feature/contacts

bortoletti-giorgia committed:
fix: some fixes after merging

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092&bid=16666723)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java:
```diff
@@ -2,45 +2,83 @@
-    public SMSContactManager(Context applicationContext) {
+    private SMSContactManager(@NonNull Context applicationContext) {
+        if(instance != null)
+            throw new RuntimeException(SINGLETON_ERROR);
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java:
```diff
@@ -13,36 +19,50 @@
  */
 public class ContactListActivity extends AppCompatActivity {
 
+    private SMSContactManager contactManager;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'contactManager' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java:
```diff
@@ -13,36 +19,50 @@
 
+    private SMSContactManager contactManager;
+    private SMSContactRecyclerAdapter recyclerAdapter;
+    private Toolbar searchToolbar;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'searchToolbar' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java:
```diff
@@ -0,0 +1,82 @@
+    private static String contactAddress;
+
+    private EditText contactNameField;
+    private TextView contactPhoneField;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'contactPhoneField' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java:
```diff
@@ -40,6 +44,7 @@
     private EditText txtPhoneNumber;
     private ImageButton sendAlarmRequestButton;
     private ImageButton sendLocationRequestButton;
+    private Toolbar mainToolbar;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'mainToolbar' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

bortoletti-giorgia committed:
fix: some fixes after Codacy's review

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092&bid=16666723)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ContactListActivity.java:
```diff
@@ -13,36 +19,49 @@
  */
 public class ContactListActivity extends AppCompatActivity {
 
+    private SMSContactManager contactManager;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'contactManager' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java:
```diff
@@ -2,45 +2,85 @@
-    public SMSContactManager(Context applicationContext) {
+    private SMSContactManager(@NonNull Context applicationContext) throws RuntimeException{
+        if(instance != null)
+            throw new RuntimeException(SINGLETON_ERROR);
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

bortoletti-giorgia committed:
fix: some fixes after Codacy's review

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092&bid=16666723)

bortoletti-giorgia committed:
fix: add ExceptionInInitializerError for Singleton in SMSContactManager

turcato-niccolo commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java:
```diff
@@ -2,45 +2,84 @@
      */
-    public SMSContactManager(Context applicationContext) {
+    private SMSContactManager(@NonNull Context applicationContext) throws ExceptionInInitializerError {
+        if(instance != null)
```
Do you need to throw the exception if the constructor is private?

bortoletti-giorgia replied:
Maybe not but it might be useful for specs or not (?)

ricdezen replied:
Technically you can break the private lock and access the constructor anyway, altough highly unlikely, I used to put this in my singletons

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092&bid=16666723)

turcato-niccolo reviewed (commented):
Works on Nexus 5, API 29

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java:
```diff
@@ -117,6 +125,53 @@ public void requestPermissions() {
+     * @param message Received SMSMessage class of SmsHandler library
+     */
+    public void onMessageReceived(SMSMessage message) {
+        manager.activeResponse(message, AlarmAndLocateResponseActivity.class);
```
Both this and `onDestroy` should not be here, they interfere with the background receival

turcato-niccolo replied:
This is a leftover from the old demos

ricdezen replied:
Yes, and they should be removed, because the service handles the receival, and must not be destroyed every time the activity is killed

bortoletti-giorgia replied:
Ok, thanks. I have removed them.

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java:
```diff
@@ -0,0 +1,81 @@
+    public void modifyContact(View view) {
+        String newContactName = contactNameField.getText().toString();
+
+        if (newContactName.equals(contactName)) {
```
Please check what happens on null text, I'm not entirely sure

turcato-niccolo replied:
Shouldn't `contactNameField.getText()` return an empty string?

bortoletti-giorgia replied:
I have tested it with empty text and the contact is updated with empty name. This is right because name is optional.

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java:
```diff
@@ -0,0 +1,81 @@
+            contactManager.modifyContactName(new SMSPeer(contactAddress), newContactName);
+            Toast.makeText(getApplicationContext(), ActivityConstantsUtils.CONTACT_UPDATED, Toast.LENGTH_LONG).show();
+        }
+        startActivity(new Intent(ModifyContactActivity.this, ContactListActivity.class));
```
You don't need to start a new activity, it's already open behind this, you should be able to call `onBackPressed` and obtain the same result

bortoletti-giorgia replied:
It's a very good advice. I changed that lines but the problem remains: when `ModifyContactActivity` is closed, I need to update the list of contacts in the `ContactListActivity`. So I have implemented the `onResume` method and notified changes at the `Adapter`. But it doesn't refresh data. Do you have some advice?

bortoletti-giorgia replied:
Ok I solved it. Now I do a _commit_ with changes.

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactConverterUtils.java:
```diff
@@ -0,0 +1,58 @@
+ *
+ * @author Giorgia Bortoletti
+ */
+public interface ContactConverterUtils<A extends Comparable<A>, P extends Peer<A>, C extends GenericContact<A, P>> {
```
I understand what you are trying to do because Room converters are required to have static methods, just keep in mind that you will never be able to consider the `SMSContactCoverterUtils` class as a `ContactConverterUtils<String, SMSPeer, SMSContact>`, because the static reference to the class can't be erased.

bortoletti-giorgia replied:
So how could I change this class or  `SMSContactCoverterUtils`  ?

ricdezen replied:
Just leave it like this, it would take many useless steps

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java:
```diff
@@ -0,0 +1,81 @@
+        setContentView(R.layout.activity_modify_contact);
+
+        contactNameField = findViewById(R.id.modify_contact_name);
+        TextView contactPhoneField = findViewById(R.id.modify_contact_address);
```
Good that you left the phone number constant

ricdezen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java:
```diff
@@ -0,0 +1,81 @@
+
+/**
+ * Activity for adding a new contact
+ *
```
Careful here.
Also I'd add info about the fact that only the name can be modified

ricdezen reviewed (changes requested):
Aside from the very few things I pointed out (in particular the MainActivity) this is quite good code, I'm honestly impressed, you have improved in both specs and the use of interfaces, good job.

turcato-niccolo commented on app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java:
```diff
@@ -0,0 +1,81 @@
+     * Setter for the contact address that cannot be changed.
+     * It is invoked by {@link SMSContactSwipeCallback}.
+     */
+    public static void setContactAddress(String contactAddress) {
```
![image](https://user-images.githubusercontent.com/56435959/75249408-90ab8880-57d6-11ea-9c6b-2c87064490e9.png)

turcato-niccolo commented on app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java:
```diff
@@ -0,0 +1,78 @@
+    /**
+     * Constructor
+     *
+     * @param adapter          to notify contacts list change
```
Try writing specifics starting with capital letters, like:
`@param adapter          `**T**`o notify contacts list change`
Someone from group 1 suggested me just that and it makes specifics easier to read

bortoletti-giorgia replied:
Thanks, I'll modify all specs.

turcato-niccolo commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java:
```diff
@@ -0,0 +1,29 @@
+    @Override
+    public boolean onQueryTextChange(String newText) {
+        recyclerAdapter.getFilter().filter(newText);
+        return false;
```
Why does this returns `false`? Also, the other methods lack documentation.

bortoletti-giorgia replied:
I'll add the specs, thanks.
It is the same return true or false because this method calls the object to filter the data. They are immediately shown on the screen without working on them.

turcato-niccolo commented on app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java:
```diff
@@ -0,0 +1,126 @@
+    @Ignore
+    @Override
+    public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
+        return false;
```
This is the callback for when the user moves an item in the list (?) Maybe we could block this action from the view's properties

bortoletti-giorgia replied:
Yes, I'll add documentation. 
In this moment this action is blocked and user can't move an item.

turcato-niccolo replied:
Good👌

turcato-niccolo reviewed (commented):
I think I've read all, nice job💪
I've left some comments and suggestions, please check them out and let me know your thoughts

bortoletti-giorgia committed:
fix: remove some methods in MainActivity

I have removed onMessageReceived and onDestroy after reviews of Turcato and De Zen.

bortoletti-giorgia committed:
style: formatting and specs

After advice from Turcato and De Zen.

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092&bid=16666723)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java:
```diff
@@ -4,14 +4,18 @@
 import androidx.core.app.ActivityCompat;
 import androidx.core.content.ContextCompat;
 
+import com.eis.smslibrary.SMSMessage;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - com.eis.smslibrary.SMSMessage.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'com.eis.smslibrary.SMSMessage'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

bortoletti-giorgia committed:
fix: fix the transition from one activity to another

After De Zen's review.

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092&bid=16666723)

RicDeZen commented on app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java:
```diff
@@ -0,0 +1,81 @@
+package ingsw.group1.findmyphone.activity;
+
+import android.content.Intent;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused import - android.content.Intent.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

RicDeZen replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'android.content.Intent'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092)

bortoletti-giorgia committed:
style: remove unused import

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092&bid=16666723)

bortoletti-giorgia committed:
fix: add label for swipe decorator

bortoletti-giorgia committed:
feat: update filter after an action on a contact

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092&bid=16666723)

bortoletti-giorgia committed:
Merge branch 'development' into feature/contacts

RicDeZen commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java  1
- app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java  2
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSearchListener.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactSwipeCallback.java  5
- app/src/main/java/ingsw/group1/findmyphone/contacts/ContactFilter.java  6
- app/src/main/java/ingsw/group1/findmyphone/activity/CreateContactActivity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5065092&bid=16666723)

ricdezen reviewed (approved):
Everything seems alright, I approve
